### ISSUE TITLE
Server-sent events don't work when serving all functions locally

### ISSUE THREAD

**Issue by sagacity-felix** [supabase/cli#1114]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

I copied the example from [Web Streams at the Edge](https://deno.com/blog/deploy-streams#server-sent-events) as seen in [this video](https://www.youtube.com/watch?v=9N66JBRLNYU). When I serve it locally using `supabase functions serve <function_name, here "stream">` and call it through the cURL, everything works well. However, when I serve all my functions together using `supabase functions serve`, the SSE function does not send any events.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a new functions using `supabase functions new stream`
2. Use the following code for the new function:
```typescript
import { serve } from "https://deno.land/std@0.140.0/http/server.ts";

const msg = new TextEncoder().encode("data: hello\r\n\r\n");

serve(async (_) => {
  let timerId: number | undefined;
  const body = new ReadableStream({
    start(controller) {
      timerId = setInterval(() => {
        controller.enqueue(msg);
      }, 1000);
    },
    cancel() {
      if (typeof timerId === "number") {
        clearInterval(timerId);
      }
    },
  });
  return new Response(body, {
    headers: {
      "Content-Type": "text/event-stream",
    },
  });
});
```
3. Serve all functions by running `supabase functions serve`
4. Call the function using cURL:
```
curl -i --location --request POST 'http://localhost:54321/functions/v1/stream' \
  --header 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6ImFub24iLCJleHAiOjE5ODM4MTI5OTZ9.CRXP1A7WOeoJeXxjNni43kdQwgnWNReilDMblYTn_I0' \
  --header 'Content-Type: application/json' \
  --data '{"name":"Functions"}'
```
5. The console will not show any response headers immediately and also no events. After a while it will say `curl: (18) transfer closed with outstanding read data remaining`. The console says `hyper::Error(IncompleteMessage)`.

## Expected behavior

I would expect the function to behave as it would behave when served via `supabase functions serve stream`, i.e., by sending response headers immediately followed by events.

## Screenshots

<img width="551" alt="Screenshot 2023-05-17 at 15 38 54" src="https://github.com/supabase/cli/assets/123091385/f54d97a1-9a46-4a2e-bc33-e4d86b71f176">
<img width="872" alt="Screenshot 2023-05-17 at 15 39 10" src="https://github.com/supabase/cli/assets/123091385/4dabfca9-dc81-4364-aa04-4b7746452890">

## System information

- OS: macOS
- Browser (if applies): using console
- Version of supabase-js: [e.g. 6.0.2]
- Version of Node.js: [e.g. 10.10.0]

**Comment by laktek** [comment]
Taking a look at this - we will release a fix within the next 24 hours.

**Comment by laktek** [comment]
This should be fixed in CLI version 1.61.3

**Comment by sagacity-felix** [comment]
Awesome, it works! Thanks a lot for the quick fix :)
